src/main/java/org/jenkinsci/plugins/workflow/support/steps/input/InputAction.java

```diff
@@ -62,8 +63,8 @@ private synchronized void loadExecutions() {
                 }
             }
             if (execution != null) {
-                // TODO JENKINS-37154 we would rather not block here
-                for (StepExecution se : execution.getCurrentExecutions(true).get()) {
+                // JENKINS-37154 sometimes we must block here in order to get accurate results
+                for (StepExecution se : execution.getCurrentExecutions(true).get(1, TimeUnit.MINUTES)) {
```

Comment:
> we'll still see issues

Possibly, but not in the case I reproduced, which is fixed by the patch to `stop`. If there are other cases that turn up, they may be noticeable by the _temporary_ hang, in which case we can look for further fixes.

> Is there any reason we'd want this to be longer than ~1 second?

The system might be heavily loaded and just need a little time. The script might be doing some calculation in the CPS VM thread. I would not want to cause the list of inputs to be incorrect merely because of some transient load.

> Would it be worth catching the TimeOutException here and returning a null or similar?

If we do time out, the exception will be caught below and `executions` will simply be empty, which should not be fatal.